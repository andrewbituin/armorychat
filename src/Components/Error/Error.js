import React, { useEffect } from 'react';

let _timeoutId;

export default function Error(props) {
  useEffect(() => {
    _timeoutId = setTimeout(() => props.close(), 20000);

    return () => {
      clearTimeout(_timeoutId)
    }
  });

  const handleClose = (e) => {
    clearTimeout(_timeoutId);
    props.close();
  }
 
  return (
    <div className="error-container">
      <p>{props.error}</p>
      {/* icon here */}
      <button type="button" onClick={handleClose}><i className="fas fa-times"/></button>
    </div>
  )
}