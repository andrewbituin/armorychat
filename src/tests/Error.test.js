import React from "react";
import Error from "../Components/Error/Error";
import { MemoryRouter } from "react-router-dom";
import toJson from "enzyme-to-json";
import { shallow } from "enzyme";

describe.only("Error component", () => {
  const props = {
    error: "Test Error message"
  };

  it("renders without crashing", () => {
    shallow(
      <MemoryRouter>
        <Error />
      </MemoryRouter>
    );
  });

  it("renders default form", () => {
    const wrapper = shallow(<Error />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("renders the UI as expected", () => {
    const wrapper = shallow(<Error {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
