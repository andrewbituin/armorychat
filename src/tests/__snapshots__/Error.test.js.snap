// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error component renders default form 1`] = `
<div
  className="error-container"
>
  <p />
  <button
    onClick={[Function]}
    type="button"
  >
    <i
      className="fas fa-times"
    />
  </button>
</div>
`;

exports[`Error component renders the UI as expected 1`] = `
<div
  className="error-container"
>
  <p>
    Test Error message
  </p>
  <button
    onClick={[Function]}
    type="button"
  >
    <i
      className="fas fa-times"
    />
  </button>
</div>
`;
